irohad/consensus/yac/impl/yac_crypto_provider_impl.cpp

```diff
@@ -41,25 +49,40 @@ namespace iroha {
       }
 
       bool CryptoProviderImpl::verify(VoteMessage msg) {
-        return iroha::verify(
-            iroha::sha3_256(
-                PbConverters::serializeVote(msg).hash().SerializeAsString())
-                .to_string(),
-            msg.signature.pubkey,
-            msg.signature.signature);
+        auto serialized =
+            PbConverters::serializeVote(msg).hash().SerializeAsString();
+        auto blob = shared_model::crypto::Blob(serialized);
+
+        return shared_model::crypto::CryptoVerifier<>::verify(
+            msg.signature->signedData(), blob, msg.signature->publicKey());
       }
 
       VoteMessage CryptoProviderImpl::getVote(YacHash hash) {
         VoteMessage vote;
         vote.hash = hash;
-        auto signature = iroha::sign(
-            iroha::sha3_256(
-                PbConverters::serializeVote(vote).hash().SerializeAsString())
-                .to_string(),
-            keypair_.pubkey,
-            keypair_.privkey);
-        vote.signature.signature = signature;
-        vote.signature.pubkey = keypair_.pubkey;
+        vote.signature = createEmptySig();
+        auto serialized =
+            PbConverters::serializeVote(vote).hash().SerializeAsString();
+        auto blob = shared_model::crypto::Blob(serialized);
```

Comment:
I think hash signing is enough here, ping @lebdron @muratovv

Reply:
Now whole whole of the vote is signed. I suggest to sign only hash inside the vote

Reply:
In this case, we can't only sign hash because the internal signature is a data and we can't lose it.